OpenRA.Mods.Cnc/Traits/EnergyWall.cs

```diff
@@ -0,0 +1,117 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.GameRules;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Support;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Cnc.Traits
+{
+	[Desc("Will open and be passable for actors that appear friendly when there are no enemies in range.")]
+	public class EnergyWallInfo : BuildingInfo, IConditionConsumerInfo, IRulesetLoaded
+	{
+		[FieldLoader.Require]
+		[WeaponReference]
+		[Desc("The weapon to attack units on top of the wall with when activated.")]
+		public readonly string Weapon = null;
+
+		[ConsumedConditionReference]
+		[Desc("Boolean expression defining the condition to activate this trait.")]
+		public readonly ConditionExpression ActiveCondition = null;
+
+		public override object Create(ActorInitializer init) { return new EnergyWall(init, this); }
+
+		public WeaponInfo WeaponInfo { get; private set; }
+
+		public void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			WeaponInfo weaponInfo;
+
+			var weaponToLower = Weapon.ToLowerInvariant();
+			if (!rules.Weapons.TryGetValue(weaponToLower, out weaponInfo))
+				throw new YamlException("Weapons Ruleset does not contain an entry '{0}'".F(weaponToLower));
+
+			WeaponInfo = weaponInfo;
+		}
+	}
+
+	public class EnergyWall : Building, IConditionConsumer, ITick, ITemporaryBlocker
+	{
+		readonly EnergyWallInfo info;
+		IEnumerable<CPos> blockedPositions;
+
+		// Initial state is active to match Building adding the influence to the ActorMap
+		// This will be updated by ConditionsChanged at actor creation.
+		bool active = true;
+
+		public EnergyWall(ActorInitializer init, EnergyWallInfo info)
+			: base(init, info)
+		{
+			this.info = info;
+		}
+
+		IEnumerable<string> IConditionConsumer.Conditions
+		{
+			get
+			{
+				if (info.ActiveCondition != null)
+					return info.ActiveCondition.Variables;
+
+				return Enumerable.Empty<string>();
+			}
+		}
+
+		void IConditionConsumer.ConditionsChanged(Actor self, IReadOnlyDictionary<string, int> conditions)
+		{
+			if (info.ActiveCondition == null)
+				return;
+
+			var wasActive = active;
+			active = info.ActiveCondition.Evaluate(conditions) > 0;
```

Comment:
Just noting for rebasing that ` > 0` is to be removed since `BooleanExpression.Evaluate(conditions)` returns a bool.